include/xtensor/xfunctorview.hpp

```diff
@@ -0,0 +1,878 @@
+     * an lvalue. It is used e.g. to access real and imaginary parts of complex expressions.
+     * expressions.
+     * xfunctorview is not meant to be used directly, but through helper functions such
+     * as \ref real or \ref imag.
+     *
+     * @tparam F the functor type to be applied to the elements of specified expression.
+     * @tparam CT the closure type of the \ref xexpression type underlying this view
+     *
+     * @sa real, imag
+     */
+    template <class F, class CT>
+    class xfunctorview : public xview_semantic<xfunctorview<F, CT>>
+    {
+    public:
+
+        using self_type = xfunctorview<F, CT>;
+        using xexpression_type = std::decay_t<CT>;
+        using semantic_base = xview_semantic<self_type>;
+        using functor_type = typename std::decay_t<F>;
+
+        using value_type = typename functor_type::value_type;
+        using reference = typename functor_type::reference;
+        using const_reference = typename functor_type::const_reference;
+        using pointer = typename functor_type::pointer;
+        using const_pointer = typename functor_type::const_pointer;
+        using size_type = typename xexpression_type::size_type;
+        using difference_type = typename xexpression_type::difference_type;
+
+        using shape_type = typename xexpression_type::shape_type;
+
+        using stepper = xfunctor_stepper<functor_type, typename xexpression_type::stepper>;
+        using const_stepper = xfunctor_stepper<functor_type, typename xexpression_type::const_stepper>;
+
+        using broadcast_iterator = xfunctor_iterator<functor_type, typename xexpression_type::broadcast_iterator>;
+        using const_broadcast_iterator = xfunctor_iterator<functor_type, typename xexpression_type::const_broadcast_iterator>;
+
+        using iterator = xfunctor_iterator<functor_type, typename xexpression_type::iterator>;
+        using const_iterator = xfunctor_iterator<functor_type, typename xexpression_type::const_iterator>;
+
+        xfunctorview(CT) noexcept;
+
+        template <class Func, class E>
+        xfunctorview(Func&&, E&&) noexcept;
+
+        template <class E>
+        self_type& operator=(const xexpression<E>& e);
+
+        template <class E>
+        disable_xexpression<E, self_type>& operator=(const E& e);
+
+        size_type size() const noexcept;
+        size_type dimension() const noexcept;
+        const shape_type& shape() const noexcept;
+
+        template <class... Args>
+        reference operator()(Args... args);
+        reference operator[](const xindex& index);
+        reference operator[](size_type i);
+
+        template <class IT>
+        reference element(IT, IT last);
+
+        template <class... Args>
+        const_reference operator()(Args... args) const;
+        const_reference operator[](const xindex& index) const;
+        const_reference operator[](size_type i) const;
+
+        template <class IT>
+        const_reference element(IT, IT last) const;
```

Comment:
This makes it look like `first` wouldn't be needed but I think it is used in the implementation